### PR TITLE
docs: improve api type

### PR DESCRIPTION
**Pull request by hengkx**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   improve api type docs.        |
| 🇨🇳 Chinese |   优化api类型定义。        |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/input/index.en-US.md](https://github.com/ant-design/ant-design/blob/improve-docs/components/input/index.en-US.md)
[View rendered components/input/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/improve-docs/components/input/index.zh-CN.md)
[View rendered components/timeline/index.en-US.md](https://github.com/ant-design/ant-design/blob/improve-docs/components/timeline/index.en-US.md)
[View rendered components/timeline/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/improve-docs/components/timeline/index.zh-CN.md)
[View rendered components/typography/index.en-US.md](https://github.com/ant-design/ant-design/blob/improve-docs/components/typography/index.en-US.md)
[View rendered components/typography/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/improve-docs/components/typography/index.zh-CN.md)

### CODE REVIEW

File: components/input/index.zh-CN.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ Input 的其他属性和 React 自带的 [input](https://facebook.github.io/reac
 
 | 参数 | 说明 | 类型 | 默认值 |
 | --- | --- | --- | --- |
-| enterButton | 是否有确认按钮，可设为按钮文字。该属性会与 `addonAfter` 冲突。 | boolean \| ReactNode | false |
+| enterButton | 是否有确认按钮，可设为按钮文字。该属性会与 `addonAfter` 冲突。 | ReactNode | false |
```
Comment (afc163) [inline components/input/index.zh-CN.md:60]: boolean 还是留着，很难从 ReactNode 里看出这个。

Comment (hengkx) [reply components/input/index.zh-CN.md:60]: image
我看了声明 `boolean` 还是 直接 写的

Comment (yoyo837) [reply components/input/index.zh-CN.md:60]: 不使用ts的会不会文档更合适？

Comment (afc163) [reply components/input/index.zh-CN.md:60]: 主要这些属性大多数都是一开始是 boolean，后来做了优化支持了 ReactNode。这样改容易把默认语义给覆盖了。

Comment (hengkx) [reply components/input/index.zh-CN.md:60]: 加个文档链接 说明 怎么样

Comment (xrkffgg) [reply components/input/index.zh-CN.md:60]: 我觉得文档中 boolean 留着比较好

Comment (hengkx) [reply components/input/index.zh-CN.md:60]: 我刚仔细看了下 文档 在说明 和 默认值 里面 基本上 也 指出了, 可以看出是 boolean 类型,

Comment (Kermit-Xuan) [reply components/input/index.zh-CN.md:60]: 之前遇到过几次，好多人不知道 ReactNode 是什么类型，觉得可以加个链接或者文档说明一下？

Comment (hengkx) [reply components/input/index.zh-CN.md:60]: 还有一个 就是 可能 会导致 用户 写出错误的写法，我改另外一个模块的时候就是 没注意 直接看的文档声明写的 ，发现完全是冗余的

**Comment by afc163** [comment]
这个类型保持原样。

**Pull request closed without merging**